src/main/java/com/ldtteam/structures/client/BlueprintRenderer.java

```diff
@@ -135,15 +148,13 @@ private void init()
      */
     public void draw(final BlockPos pos, final MatrixStack matrixStack, final float partialTicks)
```

Comment:
This is recreating the buffer every tick!
Please rework this so that it creates one buffer per use case and reuses this.

What ever a use case is might then be up to you (Blueprint + pos or as in the original only blueprint) However this is not a nice way of doing things and a lot of performance is lost here in this method!
So if this PR is targeting performance please update this method before merging, since it will likely cause structural changes in a lot of other places.

Reply:
Which buffer? I copied this from vanilla code. init() method is called when you open uncached schematic or rotate/mirror is called